lib/topology/platforms/shell.py

```diff
@@ -359,23 +510,30 @@ def __init__(
             **kwargs
         )
 
-    def _setup_shell(self):
+    def _setup_connection(self, connection=None):
```

Comment:
I agree, and I'm OK with not changing it, is just that since this method will be called for every connection, is more semantically correct to call it ``_setup_connection``.